07-errors.Rmd

```diff
@@ -0,0 +1,84 @@
+# Error messages
+
+(This guide is almost entirely asprirational; most of the bad examples come from exiting tidyverse code.)
+
+## "Must"
+
+If you can state what the user should have done to avoid the problem, start with a general statement of what was expected. Frame it possitively, and use "must". When possible, follow it up with a concise description of what was actually supplied:
+
+```{r, eval = FALSE}
+dplyr::nth(1:10, "x")
+#> Error: `n` must be a numeric vector, not a character vector.
+
+dplyr::nth(1:10, 1:2)
+#> Error: `n` must have length 1, not length 2.
+
+purrr::modify_depth(list(list(x = 1)), 3, ~ . + 1)
+#> Error: `.depth` must be less than or equal to the minimum depth; 
+```
+
+Generally 
+
+## Can't
+
+If you can't state what was expected, write a sentence that uses "can't"
+
+```{r, eval = FALSE}
+mtcars %>% pull(b)
+#> Error: Can't find column `b` in `.data`
+
+as_vector(environment())
+#> Error: Can't coerece `.x` to a vector
+```
+
+## General style
+
+Avoid pointing the user in the wrong direction by giving an opinion about the source of the error:
```

Comment:
Is this compatible with the error we give on `filter(iris, Species = "setosa")` ?